### PR TITLE
Configure drone to publish to dockerhub

### PR DESCRIPTION
**Pull request by meldafrawi**
- Publish Longhorn manager image to dockerhub when push event happens to master branch
  - image will be published with tag (master)

### CODE REVIEW

**Comment by yasker** [comment]
LGTM.

@drpebcak Can you help to double confirm? We want to automatically push the image to the rancher/longhorn-manager and other repos. It will save building time for nightly testing.

File: .drone.yml [hunk 1]
```diff
@@ -17,6 +17,26 @@ steps:
   - name: socket
     path: /var/run/docker.sock
 
+- name: docker-publish
```
Comment (drpebcak) [inline .drone.yml]: You might want to consider marking this name to reflect this is for the 'master' tag. At some point I assume we will do actual releases through here as well.

**Comment by drpebcak** [comment]
LGTM other than my small comment. @cloudnautique will have to enable this in drone-publish for it to start working when it is merged.

File: .drone.yml (hunk 1 above)
Comment (meldafrawi) [reply .drone.yml]: I've renamed it to `docker-publish-master`

**Comment by yasker** [comment]
Thanks @drpebcak 

@meldafrawi Please squash the commits into one since it served one purpose. Also, you can submit the PR for the Longhorn engine and Longhorn tests as well.

**Pull request merged**